src/commands/kv/bucket.rs

```diff
@@ -0,0 +1,103 @@
+extern crate base64;
+
+use crate::commands::kv::delete_bulk::delete_bulk;
+use crate::commands::kv::write_bulk::write_bulk;
+
+use cloudflare::endpoints::workerskv::write_bulk::KeyValuePair;
+
+use walkdir::WalkDir;
+
+use std::ffi::OsString;
+use std::fs::metadata;
+use std::path::Path;
+
+use failure::bail;
+
+use crate::terminal::message;
+
+pub fn upload(namespace_id: &str, filename: &Path) -> Result<(), failure::Error> {
+    let pairs: Result<Vec<KeyValuePair>, failure::Error> = match metadata(filename) {
+        Ok(ref file_type) if file_type.is_dir() => parse_directory(filename),
+        Ok(_file_type) => {
+            // any other file types (namely, symlinks)
+            bail!("wrangler kv:bucket upload takes a directory",)
+        }
+        Err(e) => bail!(e),
+    };
+
+    write_bulk(namespace_id, pairs?)
+}
+
+pub fn delete(namespace_id: &str, filename: &Path) -> Result<(), failure::Error> {
+    let keys: Result<Vec<String>, failure::Error> = match metadata(filename) {
+        Ok(ref file_type) if file_type.is_dir() => {
+            let key_value_pairs = parse_directory(filename)?;
+            Ok(key_value_pairs
+                .iter()
+                .map(|kv| kv.key.clone())
+                .collect::<Vec<_>>())
+        }
+        Ok(_file_type) => {
+            // any other file types (namely, symlinks)
+            bail!(
+                "{} should be a file or directory, but is a symlink",
+                filename.display()
+            )
+        }
+        Err(e) => bail!(e),
```

Comment:
We can use `failure::bail!(e)` to improve consistency within Wrangler; it was my bad for making the `use failure::bail` pattern a thing in the first place :( I have since moved to calling `failure::bail!(e)` directly.

Reply:
done.